incubator/hnc/pkg/controllers/object_controller.go

```diff
@@ -52,6 +52,7 @@ func (r *ObjectReconciler) Reconcile(req ctrl.Request) (ctrl.Result, error) {
 	resp := ctrl.Result{}
 	ctx := context.Background()
 	log := r.Log.WithValues("trigger", req.NamespacedName)
+	r.updateReconcileCounter(log, req.Namespace, true)
```

Comment:
Do you know about `defer`? Saying:

```
defer r.updateReconcileCounter(req.Namespace, false)
```

will call this function when `Reconcile` exits, which should allow you to get rid of _all_ the other calls, I think?

Reply:
ah right!